### ISSUE TITLE
2nd scoring laptop not live

### ISSUE THREAD

**Issue by VHSCSteacher** [FIRST-Tech-Challenge/scorekeeper#476]
We run two league events at the same time. We have 1 scoring laptop as the "server" and another scoring laptop that connects to the server machine's IP address. 

The 2nd laptop never has any scores showing during a the "active match" as refs are inputting scores on their tablets. The server machine has live scores showing up as soon as the ref puts it in on their tablet.

When the match timer reaches 0:00 then all the scores will populate in the "Active match" screen.

**Comment by adwiii** [comment]
Can you go into more detail about what the set up is both from the server standpoint and connected computers? Which laptop has which Match Control Page, etc?

Does everything work except for receiving in-match score updates?

**Comment by VHSCSteacher** [comment]
The server laptop has League A match control page. The "remote" laptop has League B match control page. Everything works fine on the remote laptop (randomize, start match, etc.) except for receiving in-match score updates. All League B scoring tablets, field displays etc. are all connected to the server laptop via the server laptop IP address. The server laptop is running the scoring software at localhost.

**Comment by VHSCSteacher** [comment]
Edited my original issue description after talking to the scorekeeper that is on that field. The scores in the "Active Match" only show up after the field ref has committed the score on the tablet.

**Comment by VHSCSteacher** [comment]
@adwiii We believe strongly that the cause of this issue is the API rate limiting that has been added this season. It seems like rate limits apply to anything that is not running on localhost including another machine accessing the scoring software. It doesn't seem to make a lot of sense to rate limit yourself. Would it be possible to put in some functionality for the scoring software to know when a remote scoring machine is accessing the software as opposed to a 3rd party application?

**Comment by cmacfarl** [comment]
@VHSCSteacher Rate limiting happens only on API routes. 

It's entirely possible that you've discovered an edge case.  I don't know of any regions running two leagues simultaneously and sharing a scoring server.  We'll have to do some internal investigation.   In the meantime, maybe not optimal, but do you have a laptop to run as a second server as a workaround?

**Comment by adwiii** [comment]
I have not been able to replicate the issue locally. The setup I used to try to do this:
1. On computer 1 run the scoring system and log in as local
2. On computer 1 create meet A and meet B, and generate default accounts and match schedule for both meet A and meet B.
3. On computer 2, log in as the default account for event admin
4. On computer 1 load the Match Control Page for meet A
5. On computer 2 load the Match Control Page for meet B
6. On both Match Control Pages, start the respective match 1s.
7. On separate tablets logged in as the appropriate referee accounts, enter live scores for both matches.
8. View the live scores on the Active Match tabs on both Match Control Pages

When trying the above, both Active Match tabs updated in real time during the match. In the above, computer 1 and computer 2 were both Windows 10 machines running Chrome 79.0.3945.130 and connected to the same wireless network.

A few questions:
1. Is the behavior the same if you swap which of the two computers run the scoring system?
2. What operating system and browser are the computers running?
3. What is the network setup? How is the second computer communicating with the first?
4. Can you try running an Audience Display on the computer that is not showing active matches and see if it updates even if the Match Control Page does not?

**Comment by jvens** [comment]
@cmacfarl, this exact scenario may not be common, but I believe fundamentally this would be an identical setup to how most regions would run a two division event which has been tested at this point.

Another interesting question to add to @adwiii's list would be if you were using any third party APIs such at The Orange Alliance or FTC Scores.

**Comment by VHSCSteacher** [comment]
In our setup, we use the same accounts and passwords for all events. So meet A and meet B all have the same user accounts and passwords. The scoring laptops are connected wired to the router and everything else is wireless (scoring tablets, and fields displays).

To answer your questions:
1. I have never swapped the computers, but we recently upgraded from very old laptops to brand new laptops and we still have the same issues.

2. Windows 10 and Chrome (not sure of the exact Chrome version)

3. Network setup is as described above. The 2nd laptop points its browser to the IP address of the scoring server (in this case it is 192.168.1.2), as do all other devices (scoring tablets and field displays).

4. We will try that next time.

Also, when starting the match for Event B on the 2nd laptop, we notice about a 1 second lag. In other words when the ref says "3, 2, 1, go" there is about 1 second before the start horn sounds. As a work around the scorekeeper on that laptop has been hitting "start match" when the ref says "2".

**Comment by VHSCSteacher** [comment]
> @cmacfarl, this exact scenario may not be common, but I believe fundamentally this would be an identical setup to how most regions would run a two division event which has been tested at this point.
> 
> Another interesting question to add to @adwiii's list would be if you were using any third party APIs such at The Orange Alliance or FTC Scores.

Yes, we are running FTCScores on both the server (localhost) and the 2nd laptop.

**Comment by cmacfarl** [comment]
A dual division event is slightly different.  But why would you run two
instances of FTCScores?  Don't really think it should matter but I'm
curious.

**Comment by VHSCSteacher** [comment]
FTCScores cannot upload to more than 1 event at a time. It is one event per instance of FTCScores.

**Comment by tweirtx** [comment]
I wonder if this has to do with AML not staggering the fields? The server load may be more than usual

**Comment by adwiii** [comment]
@tweirtx can you explain exactly what you mean by not staggering?

Perhaps that could account for the timing of the start being wrong, however if it is consistent which control page is live and which one is delayed, I don't expect that part is a load issue.

**Comment by tweirtx** [comment]
The two fields run as independent events, to the point where having matches on both is common (one doesn't wait for the other to finish)

**Comment by VHSCSteacher** [comment]
@adwiii I think what he means is that as stated before, we run two league events simultaneously. One field per event. So there are often two matches going on at the same time. I wouldn't think would tax the scoring software at all especially since we upgraded out laptops.
